pkg/controller/machine.go

```diff
@@ -94,15 +95,27 @@ type MetricsCollection struct {
 	NodeJoinDuration    metrics.Histogram
 }
 
-// NewMachineController returns a new machine controller
-func NewMachineController(
+// NewMachineControllerOrDie returns a new machine controller
+func NewMachineControllerOrDie(
```

Comment:
Why don't we simply make this a `func NewMachineController(...) (*Controller, error) {` and let the caller decide what they want to do?
The caller can still decide if they want to fail (log something and then exit != 0) or if they want to maybe backoff and retry, or simply continue.

Reply:
yes, good idea.